src/engine/runner/shared/docker-compose.parser.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { readFileSync, writeFileSync } from "fs";
+import { join } from "path";
+import { DockerComposeParser } from "./docker-compose.parser";
+
+let tempDocument: string;
```

Comment:
You can make it object with properties to avoid `let`.

Reply:
I don't want to avoid `let`, because I would like to explicitly tell that this variable will be overwritten instead of hiding it inside an object.

Reply:
You are right!
So if you want to have an ulimate solution and avoid mutating at all I would introduce something I genericly use in all unit tests.
```
const prepare = () => {
   const dataset = {
     path:  join(__dirname, "./tests/docker-compose.yml.dist"),
     serviceKey: "redis",
   };
   const parser = new DockerComposeParser(dataset.path);
   const beforeDocument = readFileSync(dataset.path, "utf8");
   const execute = () => {
      parser.removeService(serviceKey);
      const afterDocument = readFileSync(path, "utf8");
      writeFileSync(path, beforeDocument);
      return afterDocument;
   };
   return { dataset, execute };
}

describe("Docker-compose parser", () => {
  it("can remove redis service", () => {
    const test = prepare();

    const result = test.execute();

    result
      .split("\n")
      .forEach((value) => expect(value).not.toContain(test.dataset.serviceKey));
  });
});
```

Reply:
I don't see a reason why I should avoid mutating and run the `prepare()` method in each test case if I can do it only in `beforeAll` hook. I leave it as it is.